catalog/spatial/registry/registry-federation-admin-impl/src/test/java/org/codice/ddf/registry/federationadmin/impl/FederationAdminTest.java

```diff
@@ -0,0 +1,975 @@
+        mcard.setAttribute(RegistryObjectMetacardType.PUBLISHED_LOCATIONS, new ArrayList<>());
+        QueryResponse queryResponse = mock(QueryResponse.class);
+        when(queryResponse.getResults()).thenReturn(Collections.singletonList(new ResultImpl(mcard)));
+        when(catalogFramework.query(any(QueryRequest.class))).thenReturn(queryResponse);
+        CatalogStore store = mock(CatalogStore.class);
+        catalogStoreMap.put("myDest", store);
+
+        when(store.create(any(CreateRequest.class))).thenThrow(new IngestException("Bad Ingest"));
+
+        List<Serializable> result = federationAdmin.updatePublications("myId",
+                Collections.singletonList("myDest"));
+        verify(store, times(1)).create(any(CreateRequest.class));
+        verify(catalogFramework, times(0)).update(any(UpdateRequest.class));
+        assertThat(result.size(), is(0));
+    }
+
+    @Test
+    public void testUpdatePublicationsNewPublicationError2() throws Exception {
+        MetacardImpl mcard = new MetacardImpl(new RegistryObjectMetacardType());
+        mcard.setAttribute(RegistryObjectMetacardType.REGISTRY_ID, "myId");
+        mcard.setAttribute(RegistryObjectMetacardType.PUBLISHED_LOCATIONS, new ArrayList<>());
+        QueryResponse queryResponse = mock(QueryResponse.class);
+        when(queryResponse.getResults()).thenReturn(Collections.singletonList(new ResultImpl(mcard)));
+        when(catalogFramework.query(any(QueryRequest.class))).thenReturn(queryResponse);
+        CatalogStore store = mock(CatalogStore.class);
+        catalogStoreMap.put("myDest", store);
+        CreateResponse createResponse = mock(CreateResponse.class);
+        when(store.create(any(CreateRequest.class))).thenReturn(createResponse);
+        when(createResponse.getProcessingErrors()).thenReturn(Collections.singleton(new ProcessingDetailsImpl()));
+
+        List<Serializable> result = federationAdmin.updatePublications("myId",
+                Collections.singletonList("myDest"));
+        verify(store, times(1)).create(any(CreateRequest.class));
+        verify(catalogFramework, times(0)).update(any(UpdateRequest.class));
+        assertThat(result.size(), is(0));
+    }
+
+    @Test
+    public void testUpdatePublicationsRemovePublication() throws Exception {
+        MetacardImpl mcard = new MetacardImpl(new RegistryObjectMetacardType());
+        mcard.setAttribute(RegistryObjectMetacardType.REGISTRY_ID, "myId");
+        ArrayList<String> dest = new ArrayList<>();
+        dest.add("myDest");
+        mcard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.PUBLISHED_LOCATIONS, dest));
+        QueryResponse queryResponse = mock(QueryResponse.class);
+        when(queryResponse.getResults()).thenReturn(Collections.singletonList(new ResultImpl(mcard)));
+        when(catalogFramework.query(any(QueryRequest.class))).thenReturn(queryResponse);
+        CatalogStore store = mock(CatalogStore.class);
+        catalogStoreMap.put("myDest", store);
```

Comment:
You repeat these lines a lot. Even though it's a test, you shouldn't duplicate code. Maybe make a helper method out of this? Same with any other repeated blocks of code. That way your tests are clearer, and the differences between them are more obvious.